### PR TITLE
fix watch event behavior

### PR DESCRIPTION
**Pull request by ogidow**
Fixes: #3742

consul watch holds the index of the previous event.
If the index of this event is smaller than the index of the previous event, execute the handler once, give the event that can be acquired by the event list api, and execute handler again.
since index is calculated based on event id and event id is completely randomly generated by uuid, it is not guaranteed that the index of the new event is larger than the index of the old event.

In other words, handler will be executed multiple times at random by firing one event.
In order to prevent this problem, I monotonously increase ltime as metadata of event list api, and consider ltime with logic to initialize index to 0.

### CODE REVIEW

**Comment by ogidow** [comment]
I made a wrong account to commit so I will recreate it ...
